cddagl/ui/views/main.py

```diff
@@ -2347,6 +2348,80 @@ def copy_next_soundpack(self):
 
             self.post_extraction_step3()
 
+    def preserve_custom_fonts(self):
+        """
+        Copy over any files in the previous font directory
+        that don't already exist in the current font directory.
+
+        This assumes that fonts distributed with CDDA have already
+        been extracted into the current font directory.
+        """
+        status_bar = self.get_main_window().statusBar()
+
+        join_parts = lambda parts: Path(os.path.join(*parts))
+
+        font_locations = [
+            [self.game_dir, 'font'],        # User fonts
+            [self.game_dir, 'data', 'font'] # Game fonts
+        ]
+
+        prev_font_locations = [
+            [base, 'previous_version'] + parts for base, *parts in font_locations
+        ]
+
+        # A list of tuples with the shape (CURRENT_FONT_DIR, PREV_FONT_DIR)
+        # Only tuples that contain directories that exist in the previous version are returned
+        font_paths = list(filter(
+            lambda font_dir_map: os.path.isdir(font_dir_map[1]),
+            tuple(zip(
+                map(join_parts, font_locations),
+                map(join_parts, prev_font_locations)
+            ))
+        ))
+
+        logger.info('Scanning the following user and game font directories:')
+        logger.info('{}'.format(font_locations))
+        logger.info('{}'.format(prev_font_locations))
+
+        logger.info('Mapped current to previous font dirs:')
+        logger.info('{}'.format(font_paths))
+
+        if (any(font_paths) and self.in_post_extraction):
+            logger.debug('Restoring custom fonts')
+            status_bar.showMessage(_('Restoring custom fonts'))
+
+            for font_dir, prev_font_dir in font_paths:
+                # Skip the dir if we have nothing to restore
+                if not prev_font_dir.is_dir():
```

Comment:
Is this needed if you already filter by `os.path.isdir(font_dir_map[1])` previously when defining font_paths?

Reply:
Fixed this by removing the `filter` clause in the `font_paths` variable assignment. I think doing this explicitly here is a bit easier to follow. 👌